src/allencell_ml_segmenter/thresholding/thresholding_service.py

```
@@ -99,18 +80,29 @@ def _on_threshold_changed(self, _: Event) -> None:
             )
         else:
             thresh_function = self._threshold_image
-        for layer_name, image in layers_to_threshold.items():
+        for layer in segmentation_labels:
             # Creating helper functions for mypy strict typing
             def thresholding_task() -> np.ndarray:
-                return thresh_function(image)
+                # INVARIANT: a segmentation layer must have prob_map in its metadata if it came from our plugin
+                # so we are only supporting thresholding images that are from the plugin itself.
+                if (
+                    not isinstance(layer.metadata, dict)
+                    or "prob_map" not in layer.metadata
+                ):
+                    raise ValueError(
+                        "Layer metadata must be a dictionary containing the 'prob_map' key in order to threshold."
```

this is again to satisfy mypy. It should be impossible to reach this line of code since the `self._viewer.get_all_segmentation_labels()` function above only grabs layers with "prob_map" in its metadata, and the napari layer will always have a metadata field.

---

Makes me wonder if mypy is a net positive or not for python projects...

---

its recommended... but in spots like this i agree its just overkill. 

I also cant think of a way to rewrite this so we wouldn't need this check, if you can lmk

---

yeah it seems like this promotes 

1. not DRY code
2. not failing fast